react/features/prejoin/components/country-picker/CountryPicker.js

```diff
@@ -0,0 +1,245 @@
+import { connect } from '../../../base/redux';
+import { getCountryFromDialCodeText } from '../../utils';
+
+const PREFIX_REG = /^(00)|\+/;
+
+type Props = {
+
+    /**
+     * The country to dial out to.
+     */
+    dialOutCountry: { name: string, dialCode: string, code: string },
+
+    /**
+     * The number to dial out to.
+     */
+    dialOutNumber: string,
+
+    /**
+     * Handler used when user presses 'Enter'.
+     */
+    onSubmit: Function,
+
+    /**
+     * Sets the dial out number.
+     */
+    setDialOutNumber: Function,
+
+    /**
+     * Sets the dial out country.
+     */
+    setDialOutCountry: Function,
+};
+
+type State = {
+
+    /**
+     * If the country picker is open or not.
+     */
+    isOpen: boolean,
+
+    /**
+     * The value of the input.
+     */
+    value: string
+}
+
+/**
+ * This component displays a country picker with an input for the phone number.
+ */
+class CountryPicker extends Component<Props, State> {
+    /**
+     * A React ref to the HTML element containing the {@code input} instance.
+     */
+    inputRef: Object;
+
+    /**
+     * Initializes a new {@code CountryPicker} instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isOpen: false,
+            value: ''
+        };
+        this.inputRef = React.createRef();
+        this._onChange = this._onChange.bind(this);
+        this._onDropdownClose = this._onDropdownClose.bind(this);
+        this._onCountrySelectorClick = this._onCountrySelectorClick.bind(this);
+        this._onEntryClick = this._onEntryClick.bind(this);
+        this._onKeyPress = this._onKeyPress.bind(this);
+    }
+
+    _onChange: (Object) => void;
+
+    /**
+     * Handles the input text change.
+     * Automatically updates the country from the 'CountrySelector' if a
+     * phone number prefix is entered (00 or +).
+     *
+     * @param {Object} e - The synthetic event.
+     * @returns {void}
+     */
+    _onChange({ target: { value } }) {
+        if (PREFIX_REG.test(value)) {
+            const textWithDialCode = value.replace(PREFIX_REG, '');
+
+            if (textWithDialCode.length >= 4) {
+                const country = getCountryFromDialCodeText(textWithDialCode);
+
+                if (country) {
+                    const rest = textWithDialCode.replace(country.dialCode, '');
+
+                    this.props.setDialOutCountry(country);
+                    this.props.setDialOutNumber(rest);
+
+                    return;
+                }
+            }
+        }
+
+        this.props.setDialOutNumber(value);
+    }
+
+    _onCountrySelectorClick: (Object) => void;
+
+    /**
+     * Click handler for country selector.
+     *
+     * @param {Object} e - The synthetic event.
+     * @returns {void}
+     */
+    _onCountrySelectorClick(e) {
+        e.stopPropagation();
+
+        if (this.state.isOpen) {
```

Comment:
Can't these 2 tests be simplified with: `this.setState({ isOpen: !this.state.isOpen });` ?

Reply:
Missed this. Thanks!